psycop/common/data_structures/patient.py

```diff
@@ -89,12 +116,29 @@ def to_prediction_times(
             # 2. Return prediction sequences
             prediction_sequences.append(
                 PredictionTime(
-                    patient=self,
+                    patient_slice=patient_slice,
                     prediction_timestamp=prediction_timestamp,
-                    temporal_events=filtered_events,
                     outcome=outcome_within_lookahead,
-                    static_features=self._static_features,
                 ),
             )
 
         return prediction_sequences
+
+
+@dataclass(frozen=True)
+class PatientSlice:
+    patient: Patient
+    temporal_events: Sequence[TemporalEvent]
+
+
+def patients_to_infinite_slices(patients: Sequence[Patient]) -> Sequence[PatientSlice]:
```

Comment:
isn't this just a list comprehension? Any reason to create a function

```
patient_slices = [p.to_patient_slice() for p in patient]
```

if you do want a function then I would rename it to:
```suggestion
def patients_to_slices(patients: Sequence[Patient]) -> Sequence[PatientSlice]:
```
as infinite could make it sound like a repeating generator.

Reply:
You're completely right re: functionality!

I would love to communicate the intent; that we're not generating a sub-slice limited by time-intervals, but we're just recasting the data-type. Suggestions?

Reply:
I'll merge since this is all its waiting on, but super open to changing it in another PR 👍

Reply:
Issue here: https://github.com/Aarhus-Psychiatry-Research/psycop-common/issues/287